spec/lucky_cli/generators/validators_spec.cr

```diff
@@ -0,0 +1,53 @@
+require "../../spec_helper"
+
+describe LuckyCli::Generators::Validators do
+  context "ProjectName" do
+    hyphenate = "sector-seven"
+    special_characters = "this string is emphatic!"
+    title = "From the Earth to the Moon"
+
+    it "doesnt allow special characters" do
+      LuckyCli::Generators::Validators::ProjectName.valid?(special_characters).should be_false
+    end
+
+    it "corrects special characters" do
+      expected = "this_string_is_emphatic"
+      LuckyCli::Generators::Validators::ProjectName.sanitize(special_characters).should eq expected
+    end
+
+    it "allows hyphens" do
+      LuckyCli::Generators::Validators::ProjectName.valid?(hyphenate).should be_true
+    end
+
+    it "doesnt allow uppercase characters" do
+      LuckyCli::Generators::Validators::ProjectName.valid?(title).should be_false
+    end
+
+    it "corrects uppercase characters" do
+      expected = "from_the_earth_to_the_moon"
+      LuckyCli::Generators::Validators::ProjectName.sanitize(title).should eq expected
+    end
+  end
+
+  context "FolderName" do
+    uppercase = "UPPERCASE"
+    with_spaces = "name with spaces"
+
+    it "doesnt allow spaces" do
+      LuckyCli::Generators::Validators::FolderName.valid?(with_spaces).should be_false
+    end
+
+    it "corrects spaces" do
+      expected = "name_with_spaces"
+      LuckyCli::Generators::Validators::FolderName.sanitize(with_spaces).should eq expected
+    end
+
+    it "allows uppercase letters" do
+      LuckyCli::Generators::Validators::FolderName.valid?(uppercase).should be_true
+    end
+
+    it "doesnt correct uppercase" do
+      LuckyCli::Generators::Validators::FolderName.sanitize(uppercase).should eq uppercase
```

Comment:
In situations like this I usually extract a private method in the spec so it is easier to call and easier to rename later. Something like `private def sanitize(value)`